bi-att-flow-dev/basic/model_nqa.py

```diff
@@ -270,8 +279,12 @@ def _build_loss(self):
         loss_mask = tf.reduce_max(tf.cast(self.q_mask, 'float'), 1) # [N]
 
         # Loss
-        crossent = tf.nn.sparse_softmax_cross_entropy_with_logits(labels=self.decoder_targets, logits=self.decoder_logits_train)
-        train_loss = tf.reduce_sum(tf.reduce_sum(crossent, axis=1) * loss_mask) / tf.cast(N, dtype=tf.float32)
+        #TODO: cannot calulate loss for test since max iteration cannot be matched to test length
+        if config.mode == 'train':
+            crossent = tf.nn.sparse_softmax_cross_entropy_with_logits(labels=self.decoder_targets, logits=self.decoder_logits_train)
+            train_loss = tf.reduce_sum(tf.reduce_sum(crossent, axis=1) * loss_mask) / tf.cast(N, dtype=tf.float32)
+        else:
+            train_loss=tf.zeros([1],tf.int32)
```

Comment:
@dhruv100691 : Why this?

Reply:
test loss calculate nhi ho paa rha because...shapes of logits and labels cannot be matched, since voh maximum iterations greedy helper mein should match test label sequences always,
I didnt try to match it also..since calculating test loss is not that imp..as we already have bleu scores

Reply:
Oh true true! Ya bleu scores should suffice.